app/src/main/java/org/schabi/newpipe/util/ListHelper.java

```diff
@@ -110,6 +112,83 @@ public static int getDefaultAudioFormat(final Context context,
         }
     }
 
+    /**
+     * Return a {@link Stream} list which uses the given delivery method from a {@link Stream}
+     * list.
+     *
+     * @param streamList     the original stream list
+     * @param deliveryMethod the delivery method
+     * @param <S>            the item type's class that extends {@link Stream}
+     * @return a stream list which uses the given delivery method
+     */
+    @NonNull
+    public static <S extends Stream> List<S> keepStreamsWithDelivery(
+            @NonNull final List<S> streamList,
+            final DeliveryMethod deliveryMethod) {
+        if (streamList.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Iterator<S> streamListIterator = streamList.iterator();
+        while (streamListIterator.hasNext()) {
+            if (streamListIterator.next().getDeliveryMethod() != deliveryMethod) {
+                streamListIterator.remove();
+            }
+        }
+
+        return streamList;
+    }
+
+    /**
+     * Return a {@link Stream} list which only contains URL streams and non-torrent streams.
+     *
+     * @param streamList the original stream list
+     * @param <S>        the item type's class that extends {@link Stream}
+     * @return a stream list which only contains URL streams and non-torrent streams
+     */
+    @NonNull
+    public static <S extends Stream> List<S> removeNonUrlAndTorrentStreams(
+            @NonNull final List<S> streamList) {
+        if (streamList.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Iterator<S> streamListIterator = streamList.iterator();
+        while (streamListIterator.hasNext()) {
+            final S stream = streamListIterator.next();
+            if (!stream.isUrl() || stream.getDeliveryMethod() == DeliveryMethod.TORRENT) {
+                streamListIterator.remove();
+            }
+        }
+
+        return streamList;
+    }
+
+    /**
+     * Return a {@link Stream} list which only contains non-torrent streams.
+     *
+     * @param streamList the original stream list
+     * @param <S>        the item type's class that extends {@link Stream}
+     * @return a stream list which only contains non-torrent streams
+     */
+    @NonNull
+    public static <S extends Stream> List<S> removeTorrentStreams(
+            @NonNull final List<S> streamList) {
+        if (streamList.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final Iterator<S> streamListIterator = streamList.iterator();
+        while (streamListIterator.hasNext()) {
+            final S stream = streamListIterator.next();
+            if (stream.getDeliveryMethod() == DeliveryMethod.TORRENT) {
+                streamListIterator.remove();
+            }
+        }
+
+        return streamList;
+    }
```

Comment:
All of the methods here should be replaced with predicates and the user should be able to call `info.getVideoStreams().stream().filter(NON_TORRENT).collect(Collections.toList())`. Or something similar. But seeing `new ArrayList()` [being called each time](https://github.com/TeamNewPipe/NewPipe/blob/137c2faeacd3f2acd3cf32053c959496704b6634/app/src/main/java/org/schabi/newpipe/player/resolver/VideoPlaybackResolver.java#L73) really opens up room for errors and should be solved.